### ISSUE TITLE
Trouble reproducing the results reported in paper

### ISSUE THREAD

**Issue by AyeshaNirma** [vvanirudh/social-lstm-tf#10]
Hi Anirudh,

I used your code and trained the social model without changing any params. Training went fine . However when I try to test (using the reported protocol, so observing 8 frames predicting 12 frames ) I get way higher avg displacement error. I am assuming you might have tried reproducing the results (Average displacement error ) Table 1 in the CVPR'16 paper . Did you manage to reproduce them or closely reproduce them? 
I totally understand you are not the original author so therefore just inquiring from you in regards to your views on this. Perhaps you can share some experience of your own. 

Thanks :)

**Comment by giantleap2012** [comment]
Same here!  We are getting an average error of about 40 pixels in a 700-pixel wide image and about 3.5 meters in a dataset where we have metrical data.  So we would like to know if anyone has found parameters that reduce the error.

Thanks!

**Comment by vvanirudh** [comment]
Yes. I had issues reproducing the results of the paper. The authors don't mention the units in the paper and have been mostly unhelpful when it came to clearing my doubts regarding how they processed the datasets.

But, if you tune the hyperparameters of the model enough you get close enough results. Unfortunately, I don't have the settings that worked well. Its been a while since I worked with this code.

**Comment by AyeshaNirma** [comment]
Ah, I see. Thanks a lot for your help !. 
Its a bit surprising Alahi provides your repo in his homepage  (http://web.stanford.edu/~alahi/index.html#code)

I heard multiple people contacting them for no avail. 

Thanks once again :)

**Comment by AyeshaNirma** [comment]
actually if you use the files that you have provided 'pixel_pos_interpolate' you can basically get values similar to the one reported in Paper for all dataset 

But this is **COMPLETELY INCORRECT** basically the MAD error say on zara01 0.22 given your implementation and annotation file and the way you are computing corresponds to roughly 10 % (due to normalization), which is huge if you plot it on images, seeing qualitative results reported on the paper one is bound to doubt. 

I dont know if someone has a fix to it or some day one of the author stumble upon this page and provide some insights

**Comment by vvanirudh** [comment]
Yep. It comes back to what I have been saying before, the paper is totally ambiguous on how they compute the errors. And the fact that they don't mention the units for the errors is just mind boggling to me.

**Comment by erichhhhho** [comment]
I've contacted with Alahi, and he said that the errors unit will be in meter, but, still, I cannot reproduce the result...btw the UCY dataset didn't provide data in real world coordinate system..

**Comment by vvanirudh** [comment]
I kinda expected that the error was in metres. But as you point out, that can't be reproduced (with my code).


And true the UCY dataset didn't have ground truth distances (in metres).


On Thu, Oct 05, 2017 at 1:20am, erichhhhho < notifications@github.com [notifications@github.com] > wrote:
I've contacted with Alahi, and he said that the errors unit will be in meter, but, still, I cannot reproduce the result...btw the UCY dataset didn't provide data in real world coordinate system..

**Comment by erichhhhho** [comment]
@vvanirudh  Thx.
Btw, I think there is a little mistake about transformation world coordinate to pixel coordinate in getPixelCoordinates.m of ETH dataset.

According to the definition in this essay: [A Flexible New Technique for Camera Calibration](https://www.microsoft.com/en-us/research/publication/a-flexible-new-technique-for-camera-calibration/), **2.2 Homography between the model plane and its image**, you should multiplied the homography matrix directly with world coordinate [x,y,1], so that you can get pixel, instead of it pseudoinverse.

and [this](https://stackoverflow.com/questions/7836134/get-3d-coordinates-from-2d-image-pixel-if-extrinsic-and-intrinsic-parameters-are).

I'd just like to confirm if it is true.

**Comment by AyeshaNirma** [comment]
Actually there is a homography file that exist for UCY as well. This file is used by Yamaguchi in CVPR'11  work. In theory Alahi shoudl have used the same as well.
Zara01 and Zara02 sequences is H_Zara and H_student03 (both are in meters)

[H_zara.txt](https://github.com/vvanirudh/social-lstm-tf/files/1368374/H_zara.txt)
[H_student03.txt](https://github.com/vvanirudh/social-lstm-tf/files/1368377/H_student03.txt)

**Comment by erichhhhho** [comment]
@AyeshaNirma Thx. Could you provide the url that mention these homography matries for UCY?

**Comment by AyeshaNirma** [comment]
I got the files personally from Yamaguchi. You can request for the code for her CVPR'11 work..

**Comment by avijit9** [comment]
@AyeshaNirma Can you please tell how you are computing the world-coordinates in UCY dataset?

**Comment by AyeshaNirma** [comment]
Hope this helps

`%% input:
% locs <Nx2 input locations in image plane format: x,y>
% H <3x3 Homography matrix>

%% output:
% pts <Nx2 input locations in world plane format: x,y>

function pts=i2w(locs,H)
    P = [locs ones(size(locs,1),1)] * H';
    pts = [P(:,1)./P(:,3) P(:,2)./P(:,3)];
end`

**Comment by avijit9** [comment]
@AyeshaNirma  I just wanted to know, how to convert my average displacement error in terms of pixels to meters?

**Comment by AyeshaNirma** [comment]
Did you actually tried mapping how much pixels corresponds to meters (using the Hom.).
To answer your question, convert the annotation to world plane.

**Comment by avijit9** [comment]
@AyeshaNirma 
Yes. I have first denormalized the data and then use the code you provided to map image coordinate to world co-ordinate. Then, I am calculating the ADE metric which is in meter, right?

I have another question, the two homography matrix provided by you (which you got from Yamaguchi) are H_zara.txt and H_student03.txt. The first matrix corresponds to Zara01 and Zara02 sequence and the second one is for the UCY univ dataset?

**Comment by AyeshaNirma** [comment]
Yes
and
Yes

**Comment by framarbon** [comment]
@vvanirudh Did you use the Homography files that @AyeshaNirma commented for computing the ADE in your publication "Social Attention: Modeling Attention in Human Crowds"?

Thanks :)

**Comment by debanjan-nandi** [comment]
Hello, 
I am just following up here. Has anybody managed to replicate the results yet? Thanks in advance
